### PR TITLE
Add DatePicker component

### PR DESCRIPTION
**Pull request by rschiffer**
Contributes to story: https://trello.com/c/9q47veqZ

### CODE REVIEW

File: src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        nextMonthButtonDisabled,
+      }) => (
+        <div
+          style={{
+            margin: 10,
+            display: 'flex',
+            justifyContent: 'center',
+          }}
+        >
+          <Button
```
Comment (lukasbals) [inline src/components/DatePicker/DatePicker.tsx:69]: Can't you use `IconButton` here?

File: src/components/DatePicker/DatePicker.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import fr from 'date-fns/locale/fr';
+
+export interface DatePickerProps {
+  yearRange?: { start: number; end: number };
+  locale?: 'de' | 'en' | 'fr';
+}
+
+export const DatePicker: React.FC<DatePickerProps> = ({
+  yearRange = { start: 1990, end: new Date().getFullYear() + 100 },
+  locale = 'de',
```
Comment (lukasbals) [inline src/components/DatePicker/DatePicker.tsx:25]: The default locale should be en I think

File: src/components/DatePicker/DatePicker.tsx [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+      renderCustomHeader={({
+        date,
+        changeYear,
+        changeMonth,
+        decreaseMonth,
+        increaseMonth,
+        prevMonthButtonDisabled,
+        nextMonthButtonDisabled,
+      }) => (
+        <div
```
Comment (raulsteurer) [inline src/components/DatePicker/DatePicker.tsx:110]: Would this work?
```suggestion
        <Flex margin="10" justifyContent="center">
```

File: src/components/DatePicker/DatePicker.test.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      `${new Date().toLocaleDateString('de')}`
+    );
+  });
+
+  it('changes the date value in the field if the user clicks on another date in the calender', async () => {
+    setup();
+
+    fireEvent.click(screen.getByTestId('datepicker-input1'));
+    await waitFor(() =>
+      fireEvent.change(screen.getByTestId('datepicker-input1'), { target: { value: '10.10.2022' } })
```
Comment (raulsteurer) [inline src/components/DatePicker/DatePicker.test.tsx:29]: You could use a different number for month and day to ensure that the order does not get mixed up
```suggestion
      fireEvent.change(screen.getByTestId('datepicker-input1'), { target: { value: '20.04.2022' } })
```

**Review by raulsteurer — COMMENTED** [review]
Whoa great work!

Minor UI/UX nitpicks, but I think this should really be up to @tobsut97:
* I think a gap between the month and date selector would look good.
* Is it possible make the dates of the previous/next month gray?
* Can you use `primary.700` as the color for the selected day, so that it always matches the theme?

I visualised this real quick using the inspector tool:
image

**Review by lukasbals — COMMENTED** [review]
I think it would make sense to align the borders of the date picker with the boarders of the `Container` component 🤔

**Review by lukasbals — APPROVED** [review]
Nice work!

**Pull request merged**